holidays/countries/singapore.py

```diff
@@ -84,6 +84,10 @@ def __init__(
         super().__init__(years, expand, observed, subdiv, prov, state)
 
     def _populate(self, year) -> None:
+        def _add_holiday(dt: date, hol: str) -> None:
```

Comment:
The name could be just `add_holiday(dt: date, hol: str) -> None:` as that's an inner function with no name conflicts.

Reply:
Many countries with islamic holidays have this function exactly as `_add_holiday`, so I think it's better to be uniformly for now.

Reply:
Agree w @KJhellico , the underscore suggests it's a private member 👍